Objects/bytes_methods.c

```diff
@@ -556,11 +556,56 @@ find_internal(const char *str, Py_ssize_t len,
     return res;
 }
 
+#define FIND_CHUNK_SIZE 10000
+#define RFIND_CHUNK_SIZE FIND_CHUNK_SIZE
+
 PyObject *
-_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *sub,
+_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *subobj,
                Py_ssize_t start, Py_ssize_t end)
 {
-    Py_ssize_t result = find_internal(str, len, "find", sub, start, end, +1);
+    Py_ssize_t result;
+    if (PyTuple_Check(subobj)) {
+        result = -1;
+        ADJUST_INDICES(start, end, len);
+        // Work in chunks
+        for (; result == -1 && start <= end; start += FIND_CHUNK_SIZE) {
+            Py_ssize_t cur_end = start + FIND_CHUNK_SIZE - 1;
+            for (Py_ssize_t i = 0; i < PyTuple_GET_SIZE(subobj); i++) {
+                PyObject *subseq = PyTuple_GET_ITEM(subobj, i);
+                Py_ssize_t sub_len;
+                Py_buffer subbuf;
+                if (!PyObject_CheckBuffer(subseq)) {
+                    sub_len = 1;
+                }
+                else if (PyObject_GetBuffer(subseq, &subbuf,
+                                            PyBUF_SIMPLE) != 0)
+                {
+                    return NULL;
+                }
+                else {
+                    sub_len = subbuf.len;
+                }
+                Py_ssize_t sub_end = cur_end + sub_len;
+                if (sub_end > end) {
+                    sub_end = end;
+                }
+                Py_ssize_t new_result = find_internal(str, len, "find", subseq,
```

Comment:
```suggestion
        if (sub_len >= sub_end - start) {
                Py_ssize_t new_result = find_internal(str, len, "find", subseq,
                # Rest of loop
        }
```

Reply:
I don't want to do unnecessary micro-optimisations. The checks currently present were necessary anyway, so it didn't matter to modify them a little bit. e.g.

```c
cur_end = new_result;
if (cur_end  != start) {
    cur_end--;
}
```

Was modified to:

```c
if (new_result == start) {
    return start;
}
cur_end = new_result - 1;
```